source/PlayState.hx

```diff
@@ -1,29 +1,162 @@
-		var myMsg = new FlxText(125, 200, 0);
-		myMsg.text = "Greetings Humans.";
-		myMsg.setFormat("assets/font.ttf", 50, FlxColor.RED, CENTER);
-		myMsg.setBorderStyle(OUTLINE, FlxColor.BLUE, 1);
-		add(myMsg);
 
-		var player:Hero = new Hero();
+		hero = new Hero();
+		hero.x= 240;
+		hero.y= 170;
+		add(hero);
+
+		initializeWalls();
+		initializeFireballs();
+		initializeCoins();
+	}
+
+	private function initializeWalls() {
+		walls = new FlxTypedGroup<Wall>();
+
+		for (i in 0...WALL_COUNT) {
+			var x:Float = WALL_START_X + (i * Wall.WIDTH);
+			var y:Float = WALL_START_Y;
+			var wall:Wall = new Wall(x, y);
+			walls.add(wall);
+		}
+		add(walls);
+	}
+	private function initializeCoins() {
+		coins = new FlxTypedGroup<Coin>();
+
+		for (i in 0...COIN_COUNT)
+			{
+				var x:Float = FlxG.random.int(COIN_SPAWN_BORDER,
+					 FlxG.width - COIN_SPAWN_BORDER);
+				var y:Float = FlxG.random.int(COIN_SPAWN_BORDER,
+					 FlxG.height - COIN_SPAWN_BORDER);
+				var coin = new Coin(x , y);
+				coins.add(coin);
+			}
+		add(coins);
+	}
 
-		add(player);
+	private function initializeFireballs() {
+		fireballs = new FlxTypedGroup<Fireball>();
 
+		for (i in 0...FIREBALL_COUNT) {
+			var x:Float = FlxG.random.int(FIREBALL_SPAWN_BORDER, 
+				FlxG.width - FIREBALL_SPAWN_BORDER);
+			var y:Float = FlxG.random.int(FIREBALL_SPAWN_BORDER, 
+				FlxG.height - FIREBALL_SPAWN_BORDER);
+			var fireball = new Fireball(x, y);
+			fireballs.add(fireball);
+		}
+		add(fireballs);
 	}
 
-	override public function update(elapsed:Float)
+	override public function update(elapsed:Float):Void
 	{
 		super.update(elapsed);
-		
+
+		// Uncomment to collide hero against all wall objects.
+		FlxG.collide(hero, walls);
+
+		// Uncomment to trigger custom logic when a player overlaps with a fireball.
+		 FlxG.overlap(hero, fireballs, resolveHeroFireballOverlap);
+
+		FlxG.overlap(hero, coins, reseolveHeroCoinOverlap);
+
+		// Wrap various objects if gone offscreen.
+		screenWrapObject(hero);
+		for (fireball in fireballs) {
+			screenWrapObject(fireball);
+		}
+		for (coin in coins) {
+			screenWrapObject(coin);
+		}
+
+	}
+
+	private function screenWrapObject(wrappingObject:FlxObject) {
+		if (wrappingObject.x > FlxG.width) {
+			wrappingObject.x = 0 - wrappingObject.width;
+		} else if (wrappingObject.x + wrappingObject.width < 0) {
+			wrappingObject.x = FlxG.width;
+		}
+
+		if (wrappingObject.y > FlxG.height) {
+			wrappingObject.y = 0 - wrappingObject.height;
+		} else if (wrappingObject.y + wrappingObject.height < 0) {
+			wrappingObject.y = FlxG.height;
+		}
+	}
+
+	/**
+		Function called when an overlap between hero and fireball is detected.
+	**/
+	private function resolveHeroFireballOverlap(hero:Hero, fireball:Fireball) {
+		trace("Hero and Fireball collided!");
+		fireball.kill();
+		hero.kill();
+
+		loseMessage = new FlxText(200, 200, 0, "You Died!", 50);
+		loseMessage.color.red;
+		loseMessage.setBorderStyle(SHADOW, FlxColor.CYAN);
+		add(loseMessage);
```

Comment:
Would be nice to have some extra logic to allow the player to restart the state if they'd like to try again (saves them the time of closing and reopening everything)

Same applies to the "win" message. No changes required here, just a comment.